merlin/core/compat/__init__.py

```diff
@@ -21,7 +21,7 @@
 
 from merlin.core.has_gpu import HAS_GPU  # noqa pylint: disable=unused-import
 
-if not cuda.is_available():
+if not HAS_GPU:
     cuda = None
```

Comment:
Thanks @jperez999 !

I think we should also remove the use of `numba.cuda` below. In `device_mem_size`, we `try` to use `cuda.current_context().get_memory_info` to query the memory size. However, this creates a cuda context and returns exactly the same information as `pynvml_mem_size` anyway:


```python
    try:
        if kind == "free":
            return int(cuda.current_context().get_memory_info()[0])
        else:
            return int(cuda.current_context().get_memory_info()[1])
    except NotImplementedError:
        if kind == "free":
            # Not using NVML "free" memory, because it will not include RMM-managed memory
            warnings.warn("get_memory_info is not supported. Using total device memory from NVML.")
        size = pynvml_mem_size(kind="total", index=0)
        return size
```

Note that we were originally using `rmm.get_info` in the `try` block (I think), because that method used to tell us how much of the allocated memory pool was also free. However, rmm deprecated that method, because it became difficult to support. Using `pynvml` and `numba.cuda` will both the rmm pool allocation as "used" memory, and so there is no good reason to use `numba.cuda` over `pynvml` (in fact, it seems that it breaks `dask_cuda`!)

Disclaimer: I'm only 90% sure of what I'm saying. So, please feel free to push back if you think I'm wrong.

Reply:
Ah, just realized @pentschev [already called out the same thing](https://github.com/NVIDIA-Merlin/core/pull/364#issuecomment-1888610944). Thanks Peter!

Reply:
@rjzamora  I have fixed this issue and removed cuda from use. We are still using numba in other parts of the merlin ecosystem so I wont completely remove it. But if its not here during initialization, we should be good to allow dask to create the context worry free.